### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by vivek-zemoso**
## Added the code Unit Testing.

1. Added Test Cases for Atoms and Molecules. 
2. Sonar Cloud report shows coverage of **85%**. 
3. After this PR gets merged into `main`, all the unit test cases will be available in `main branch`

### Please review the code and suggest feedback, if any !!

### CODE REVIEW

File: src/components/atoms/icon/__snapshots__/index.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (Nishanth-17) [inline src/components/atoms/icon/__snapshots__/index.test.tsx.snap:1]: Don't commit .snap files

File: package-lock.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
         "@storybook/preset-create-react-app": "^4.0.1",
         "@storybook/react": "^6.4.14",
         "babel-loader": "^8.2.3",
+        "jest-sonar-reporter": "^2.0.0",
```
Comment (Nishanth-17) [inline package-lock.json:41]: Do not commit package-lock.json

**Review by Nishanth-17 — COMMENTED** [review]
Resolve the comments

**Comment by vivek-zemoso** [comment]
For both of the suggestions, It was actually discussed in the LDP morning standup, and the mentor there said that we should commit `package-lock.json` file. 

About the snapshots, the mentor in the LDP meet said to commit the snapshots along with the code.

**Pull request merged**